### PR TITLE
Replace AppEngineCondition for ConditionalOnGcpEnvironment

### PR DESCRIPTION
**Pull request by eddumelendez**
`ConditionalOnGcpEnvironment` allows to validate against which GCP
environment the application is running, so the use of this annotation
with the proper values should be enough. Also, `AppEngineCondition` class is removed.

Fixes gh-1254

### CODE REVIEW

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java [hunk 1]
```diff
@@ -77,7 +78,7 @@
 	 * The AppEngine Configuration for the {@link AppEngineCloudSqlJdbcInfoProvider}.
 	 */
 	@ConditionalOnClass(com.google.cloud.sql.mysql.SocketFactory.class)
-	@Conditional(AppEngineCondition.class)
+	@ConditionalOnGcpEnvironment({GcpEnvironment.APP_ENGINE_FLEXIBLE, GcpEnvironment.APP_ENGINE_STANDARD})
```
Comment (meltsufin) [inline spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java]: The original condition would only match for App Engine Standard, not Flex.
Also, I believe that with the latest version of the Cloud SQL Socket Factory, you don't even need special configuration for App Engine Standard, but I haven't validated that.

Comment (eddumelendez) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java]: thanks @meltsufin still trying to resolve the issue I have with the usage of `@ConditionalOnGcpEnvironment`.

> Also, I believe that with the latest version of the Cloud SQL Socket Factory, you don't even need special configuration for App Engine Standard, but I haven't validated that.

Are there any way I can validate this?

Thanks in advance.

Comment (meltsufin) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java]: @kurtisvg Can you confirm that the JDBC connection string format is the same now for both App Engine Standard and Flex/GKE/etc.

Comment (kurtisvg) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java]: Correct - if you are using the SocketFactory, the JDBC url should be the same regardless of environment.

**Comment by meltsufin** [comment]
Made changes and verified that it still works on App Engine Standard.

**Comment by elefeint** [comment]
Master build is broken; let me look into that before merging.

**Pull request merged**